wrappers/wasm/BarcodeReader.cpp

```diff
@@ -49,9 +44,18 @@ std::vector<ReadResult> readBarcodes(ImageView iv, bool tryHarder, const std::st
 		std::vector<ReadResult> readResults{};
 		readResults.reserve(results.size());
 
+		thread_local const emscripten::val Uint8Array = emscripten::val::global("Uint8Array");
+
 		for (auto&& result : results) {
-			readResults.push_back({ToString(result.format()), result.text(), std::string(result.bytes().asString()),
-								   ToString(result.error()), result.position(), result.symbologyIdentifier()});
+			ByteArray bytes = result.bytes();
```

Comment:
Very nice! I have one tiny improvement to suggest: this line makes a copy of result.bytes(), which is unnecessary. Either use `result.bytes().data()` below or change this line to `const ByteArray& bytes = result.bytes()`. I have not tested your changes myself, will do.

Reply:
Thanks, fixed now.